### ISSUE TITLE
Organization with no activities - on prod cannot add activity

### ISSUE THREAD

**Issue by dragonflypl** [krico/jas#407]
On test env, request for org with no activities returns nice array:

```
{
  "items" : [ ]
}
```

whereas on prod:

```
{
 "kind": "jasify#activitiesItem",
 "etag": "\"4-u1xgOcBIjPWmvLOAM65d6-Yig/J9aiACwofFqX_M2jAfzGoV1Yass\""
}
```

In controller I expect items to be present in the reponse, which caused this issue....

We don't have etags implemented on client side, don't we?

**Comment by pauljanes** [comment]
Yeah i just noticed this bug as in prod I created a new org called ZeoT (Zentrum für orientalische Tanzkunst), but now I cannot add any activities to this org https://jasify-schedule.appspot.com/#/admin/activities/O3020001

**Comment by krico** [comment]
Unfortunately I have encountered a few scenarios where the devserver **does not** behave like the production server, especially when it comes to data validation.  This seems to be one of those cases.

I went back to one of the sample apps provided by google, and their function that handles list results (i.e. with `result.items`) are written like so:

```javascript
      function(resp) {
        if (!resp.code) {
          resp.items = resp.items || [];
          for (var i = 0; i < resp.items.length; i++) {
            google.devrel.samples.hello.print(resp.items[i]);
          }
        }
      }
```

We use the "promise" style with `.then(success, failure)` so we don't have to check resp.code, but we still need to check resp.items.

I guess ideally our services (activity.service.js) are the best place to handle that and guarantee the result is always defined when the controller calls.

I will give it a go for our existing services .

For your reference, the other thing that is not the same is not sending values that are not marked as `@Nullable` in the java api code. That works in dev and fails in prod.
